packages/synapse-react-client/src/components/SynapseTable/SynapseTable.tsx

```diff
@@ -66,26 +73,32 @@ const columnHelper = createColumnHelper<Row>()
 export function SynapseTable(props: SynapseTableProps) {
   const { showDownloadColumn, hideDownload } = props
   const {
+    rowSet,
+    isLoadingNewPage,
     showAccessColumn,
     showDirectDownloadColumn = showDownloadColumn,
     hideAddToDownloadListColumn = hideDownload,
     columnLinks,
   } = props
-  const { getCurrentQueryRequest } = useQueryContext()
+  const {
+    entityId,
+    versionNumber,
+    getCurrentQueryRequest,
+    fileIdColumnName,
+    fileVersionColumnName,
+  } = useQueryContext()
   const queryRequest = useMemo(
     () => getCurrentQueryRequest(),
     [getCurrentQueryRequest],
   )
-  const data = useAtomValue(tableQueryDataAtom)
-  const isLoadingNewBundle = useAtomValue(isLoadingNewBundleAtom)
-  const entity = useAtomValue(tableQueryEntityAtom)
+
+  const selectColumns = rowSet.headers
```

Comment:
Get the selectColumns from the rowSet, so we no longer have to wait for `queryMetadata` to resolve to show the table!

Reply:
This seems like a significant win.  What is the behavior for real-world query UX?

Reply:
For very large tables & views, where the stats take a long time to compute, the page of row data is shown to the user up to seconds before the rest of the UI.